### PR TITLE
Removes vertex-format type fix

### PR DESCRIPTION
**Pull request by kpal81xd**
Removes vertex-format type fix and replaces description object with typedef of `AttributeDescription`
Requires `api-reference` PR to be merged in conjunction https://github.com/playcanvas/api-reference/pull/14

- [x] TS linter
- [x] API reference

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by kpal81xd** [comment]
> So the types look correct if you remove this? Have you visual diffed the vertex-format.d.ts before and after?

I had forgotten to add the actual changes to the commit 😅

**Comment by willeastcott** [comment]
Have you also examined how this affects the docs?

**Comment by willeastcott** [comment]
Ah yeah, goes from:

image

To:

image

File: src/platform/graphics/vertex-format.js [hunk 1]
```diff
@@ -63,14 +63,11 @@ const deviceCache = new DeviceCache();
  * @property {number} elements[].size The size of the attribute in bytes.
  * @category Graphics
  */
+
```
Comment (willeastcott) [inline src/platform/graphics/vertex-format.js:66]:
```suggestion
```

**Review by willeastcott — APPROVED** [review]

**Pull request merged**